cmd/cosign/cli/public_key.go

```diff
@@ -54,12 +61,39 @@ EXAMPLES
 			if (*key == "" && *kmsVal == "") || (*key != "" && *kmsVal != "") {
 				return &KeyParseError{}
 			}
-			return GetPublicKey(ctx, *key, *kmsVal, GetPass)
+			// Get private key file.
+			var reader io.Reader = (*os.File)(nil)
+			if *key != "" {
+				cl := filepath.Clean(*key)
+				var err error
+				reader, err = os.Open(cl)
+				if err != nil {
+					return err
+				}
+			}
+
+			writer := NamedWriter{Name: "", Writer: nil}
+			var f *os.File
+			// Open output file for public key if specified.
+			if *outFile != "" {
+				writer.Name = *outFile
+				var err error
+				f, err = os.OpenFile(*outFile, os.O_WRONLY|os.O_CREATE, 0600)
+				if err != nil {
+					return err
+				}
+				writer.Writer = f
+				defer f.Close()
```

Comment:
I read online that you probably shouldn't defer close on write-able files. Can fix if that's the case.